tests/integration/test_backup_restore_on_cluster/test.py

```diff
@@ -1016,29 +1102,50 @@ def test_table_in_replicated_database_with_not_synced_def():
 
 
 def has_mutation_in_backup(mutation_id, backup_name, database, table):
-    return (
-        os.path.exists(
-            os.path.join(
-                get_path_to_backup(backup_name),
-                f"shards/1/replicas/1/data/{database}/{table}/mutations/{mutation_id}.txt",
-            )
+    metadata_path = os.path.join(get_path_to_backup(backup_name), ".backup")
+    with open(metadata_path, "r") as f:
+        xml_root = ET.fromstring(f.read())
+
+    data_file_name_generator_elem = xml_root.find("data_file_name_generator")
+    data_file_name_generator = (
+        data_file_name_generator_elem.text
+        if data_file_name_generator_elem is not None
+        else "first_file_name"
+    )
+
+    contents = xml_root.find("contents")
+    file_map = {}
+    if contents is not None:
+        for file_elem in contents.findall("file"):
+            name_elem = file_elem.find("name")
+            data_elem = file_elem.find("data_file")
+            if name_elem is not None and data_elem is not None:
+                file_map[name_elem.text] = data_elem.text
+
+    for replica in [1, 2, 3]:
+        file_name = f"shards/1/replicas/{replica}/data/{database}/{table}/mutations/{mutation_id}.txt"
+        data_file_name = (
+            file_name
+            if data_file_name_generator == "first_file_name"
+            else file_map.get(file_name)
         )
-        or os.path.exists(
-            os.path.join(
-                get_path_to_backup(backup_name),
-                f"shards/1/replicas/2/data/{database}/{table}/mutations/{mutation_id}.txt",
-            )
-        )
-        or os.path.exists(
-            os.path.join(
-                get_path_to_backup(backup_name),
-                f"shards/1/replicas/3/data/{database}/{table}/mutations/{mutation_id}.txt",
-            )
-        )
-    )
+
+        if data_file_name and os.path.exists(
+            os.path.join(get_path_to_backup(backup_name), data_file_name)
+        ):
+            return True
+
+    return False
 
 
-def test_mutation():
+@pytest.mark.parametrize(
+    "name_gen",
+    [
+        pytest.param("", id="first_file_name"),
+        pytest.param("checksum"),
+    ],
+)
+def test_mutation(name_gen):
```

Comment:
Test `test_mutation` isn't good and needs to be rewritten because it doesn't check that mutations are restored. I think it's better to keep it as is for now.